src/libAtomVM/memory.h

```diff
@@ -343,9 +343,9 @@ static inline void memory_destroy_heap_fragment(HeapFragment *fragment)
 /**
  * @brief Destroy a root heap. First sweep its mso list.
  */
-static inline void memory_destroy_heap(Heap *heap, GlobalContext *global)
+static inline void memory_destroy_heap(Heap *heap, GlobalContext *global, bool from_task)
```

Comment:
Coding style: I usually suggest avoiding using booleans for things that are not clearly predicates.

E.g. Let's suppose having a draw_text function, having `draw_text("Hello", Bold)` is way more understandable than having `draw_text("Hello", true)`.

In this specific case I even suggest keeping `memory_destroy_heap` as is, and I would add a `memory_destroy_heap_from_task`, also for consistency reasons with all the newly introduced functions.